auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,405 @@
+                _(messageTitle), _(
+                    messageContent + "%s") %
+                tools.ustr(e))
+        else:
+            raise Warning(_(messageTitle), _(messageContent))
+
+    @api.model
+    def schedule_backup(self):
+
+        for rec in self.search([]):
+            db_list = self.get_db_list(rec.host, rec.port)
+            if rec.name in db_list:
+                file_path = ''
+                bkp_file = ''
+                try:
+                    if not os.path.isdir(rec.bkp_dir):
+                        os.makedirs(rec.bkp_dir)
+                except:
+                    raise
+                # Create name for dumpfile.
+                bkp_file = '%s_%s.dump.zip' % (
+                    time.strftime('%d_%m_%Y_%H_%M_%S'),
+                    rec.name)
+                file_path = os.path.join(rec.bkp_dir, bkp_file)
+                conn = self.get_connection(rec.host, rec.port)
+                bkp = ''
+                try:
+                    bkp = execute(
+                        conn, 'dump', tools.config['admin_passwd'], rec.name)
+                except:
+                    _logger.notifyChannel(
+                        'backup', netsvc.LOG_INFO,
+                        _(
+                            "Couldn't backup database %s. "
+                            "Bad database administrator"
+                            "password for server running at http://%s:%s"
+                        ) % (rec.name, rec.host, rec.port))
+                    return False
+                bkp = base64.decodestring(bkp)
+                fp = open(file_path, 'wb')
+                fp.write(bkp)
+                fp.close()
+            else:
+                _logger.notifyChannel(
+                    'backup', netsvc.LOG_INFO,
+                    "database %s doesn't exist on http://%s:%s" %
+                    (rec.name, rec.host, rec.port))
+                return False
+            # Check if user wants to write to SFTP or not.
+            if rec.sftpwrite is True:
+                try:
+                    # Store all values in variables
+                    dir = rec.bkp_dir
+                    pathToWriteTo = rec.sftppath
+                    ipHost = rec.sftpip
+                    portHost = rec.sftpport
+                    usernameLogin = rec.sftpusername
+                    passwordLogin = rec.sftppassword
+                    # Connect with external server over SFTP
+                    srv = pysftp.Connection(
+                        host=ipHost,
+                        username=usernameLogin,
+                        password=passwordLogin,
+                        port=portHost)
+                    # Move to the correct directory on external server. If the
+                    # user made a typo in his path with multiple slashes
+                    # (/odoo//backups/) it will be fixed by this regex.
+                    pathToWriteTo = re.sub('([/]{2,5})+', '/', pathToWriteTo)
```

Comment:
I'm pretty sure you do want this: your code handles 5 consecutive slashes, 7 consecutive slashes, but not 6 :smile_cat: